sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryInterfaceGenerator.kt

```diff
@@ -33,26 +37,35 @@ class QueryInterfaceGenerator(val query: NamedQuery) {
         .addModifiers(DATA)
         .addSuperinterface(ClassName(query.select.sqFile().packageName, query.name.capitalize()))
 
-    var propertyPrints = listOf<String>()
+    val propertyPrints = CodeBlock.builder()
+    propertyPrints.beginControlFlow("buildString")
+    propertyPrints.addStatement("appendln(%S)", "${query.name.capitalize()}.$IMPLEMENTATION_NAME [")
+
     val constructor = FunSpec.constructorBuilder()
+    val contentToString = MemberName("kotlin.collections", "contentToString")
 
     query.resultColumns.forEach {
       typeSpec.addProperty(PropertySpec.builder(it.name, it.javaType, OVERRIDE)
           .initializer(it.name)
           .build())
       constructor.addParameter(it.name, it.javaType, OVERRIDE)
 
-      propertyPrints += "  ${it.name}: ${"$"}${it.name}"
+      propertyPrints.addStatement("appendln(%P)", buildCodeBlock {
+        add("  ${it.name}: ")
+        if (it.javaType == ByteArray::class.asTypeName()) {
```

Comment:
I tried handling all array types initially but it looked like other types required custom adapters, so I wasn't sure whether it's enough to just special-case `ByteArray`.

Reply:
`contentToString` is defined for all primitive arrays and `Array<T>`: http://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/content-to-string.html

Reply:
Yeah, I meant that when I tried handling all types of array I noticed that all but `ByteArray` are wrapped into `ColumnAdapter`s, hence `javaType` wouldn't be an array type. I haven't spent enough time to try to understand why that's case, so maybe I'm wrong, I'll take a closer look at it.